File: frontend/src/features/brand/services/brandService.ts
import axios from "axios";

import type {
  BrandProfile,
} from "../types/brand";

const API_URL =
  import.meta.env.VITE_API_URL ??
  "http://127.0.0.1:8000";


export async function getBrandProfile(
  workspaceId: string,
) {

  const response =
    await axios.get(
      `${API_URL}/brand/${workspaceId}`,
    );

  return response.data;
}


export async function createBrandProfile(
  data: BrandProfile,
) {

  const response =
    await axios.post(
      `${API_URL}/brand`,
      data,
    );

  return response.data;
}


export async function updateBrandProfile(
  workspaceId: string,
  data: BrandProfile,
) {

  const response =
    await axios.put(
      `${API_URL}/brand/${workspaceId}`,
      data,
    );

  return response.data;
}

File: backend/app/services/brand_service.py
from app.db.supabase import supabase


def create_brand_profile(data):

    response = (
        supabase
        .table("brand_profiles")
        .insert(data)
        .execute()
    )

    return response.data


def get_brand_profile(
    workspace_id: str,
):

    response = (
        supabase
        .table("brand_profiles")
        .select("*")
        .eq(
            "workspace_id",
            workspace_id,
        )
        .limit(1)
        .execute()
    )

    if not response.data:
        return None

    return response.data[0]


def update_brand_profile(
    workspace_id: str,
    data,
):

    response = (
        supabase
        .table("brand_profiles")
        .update(data)
        .eq(
            "workspace_id",
            workspace_id,
        )
        .execute()
    )

    return response.data


def get_brand_profile_for_agent(
    workspace_id: str,
):

    response = (
        supabase
        .table("brand_profiles")
        .select("*")
        .eq(
            "workspace_id",
            workspace_id,
        )
        .limit(1)
        .execute()
    )

    if not response.data:
        return {}

    return response.data[0]